### ISSUE TITLE
manage.py inspectdb: All redshift tables fail with unable to inspect table - function unnest(int2vector) does not exist

### ISSUE THREAD

**Issue by davidthewatson** [jazzband/django-redshift-backend#13]
I have:

    17:51 $ pip freeze | grep redshift
    django-redshift-backend==0.5

and:

    17:59 $ pip freeze | grep Django
    Django==1.10.3

I am running the following command:

    python manage.py inspectdb

And I get the following output on every table in my redshift schema:

    # Unable to inspect table 'dogs'
    # The error was: function unnest(int2vector) does not exist
    HINT:  No function matches the given name and argument types. You may need to add explicit type     casts.

It seems like the driver is falling back to postgres 9.x despite my database settings:

    DATABASES = {
        'default': {
            'ENGINE': 'django_redshift_backend',
            'NAME': 'dogs',
            'USER': 'user',
            'PASSWORD': 'password',
            'HOST': 'hostname',
            'PORT': '5439',
        }
    }

**Comment by schatzistogias** [comment]
Same issue here

**Comment by dohq** [comment]
same problem

**Comment by Leekyungsoo** [comment]
same problem

**Comment by gauravnexus** [comment]
Any update on this issue?

**Comment by epineda** [comment]
I'm getting:

`# The error was: function unnest(smallint[]) does not exist`

I'm using django-redshift-backend  version 0.9.1

Please, any news on this issue?

**Comment by Surgo** [comment]
Django currently supports [PostgreSQL >= 9.4](https://github.com/django/django/blob/2.1.5/docs/releases/2.1.txt#L266-L270) and Redshift's API are 8.x based.
https://github.com/django/django/blob/2.1/django/db/backends/postgresql/introspection.py#L143 `inspectdb` run `unnest(c.conkey)` and that not worked on Redshift.

**Comment by hlodver** [comment]
Based on detailed information from @Surgo (thanx!)  I used a workaround by using older django version to do the inspection. I have A LOT of tables (and am lazy) so here it is:

```bash
mkdir tmp && cd tmp
virtualenv -p python2 venv
source venv/bin/activate
pip install django==1.4.0 psycopg2
django-admin.py startproject djangold
cd djangold
export PYTHONPATH=$(pwd)
export DJANGO_SETTINGS_MODULE=djangold.settings
# Update your settings to include 
# DATABASES = {
#    'default': {
#        'ENGINE': 'django.db.backends.postgresql_psycopg2',
#        'NAME': 'db_name',
#        'USER': 'username',
#        'PASSWORD': 'pwd',
#        'HOST': 'host',
#        'PORT': '5439',
#        'OPTIONS': {
#            'options': '-c search_path=schema_name',  # if using schema
#        }
#    },
#}
# Voila! Now it works
django-admin.py inspectdb
```

I hope this helps: @epineda @gauravnexus

**Comment by aroxby-kinnek** [comment]
For anyone getting:
```
TypeError: sequence index must be integer, not 'slice'
```
I was able to get around this by going all the way back to Django 1.2

**Comment by RicoChivela1** [comment]
Unable to inspect table 'auth_group' - Can anyone help with this mistake?

**Comment by aroxby-kinnek** [comment]
@RicoChivela1, this message typically indicates that the table is missing.  Did you run `./manage.py migrate` or `./manage.py syncdb`?  If you're unfamiliar with Django I strongly recommend working through the [tutorial](https://docs.djangoproject.com/en/2.2/intro/tutorial01/).

**Comment by utapyngo** [comment]
I am getting
```
# Unable to inspect table 'table_name'
# The error was: syntax error at or near "WITH"
LINE 6:                     FROM unnest(c.conkey) WITH ORDINALITY co...
```
on
```
Django==2.2.5
django-redshift-backend==1.1.0
```

**Comment by miaoz2001** [comment]
Same issue here...Based on @hlodver 's solution (thx!) and I tested that the last working version of django is 1.7.11 which is slightly better than django==1.4 for me, in the way that 1.7.11 added one extra line 
```
class Meta:
        managed = False
```
just for anyone who needs this...

**Comment by shimizukawa** [comment]
fixed by #90 . Thanks to @MattFisher !

**Comment by shimizukawa** [comment]
It has been released https://pypi.org/project/django-redshift-backend/3.0.0/
